File: 3162/main.go
package main

import "fmt"

func main() {
	fmt.Println(numberOfPairs([]int{1,3,4}, []int{1,3,4}, 1))	// 5
	fmt.Println(numberOfPairs([]int{1,2,4,12}, []int{2,4}, 3))	// 2
}

func numberOfPairs(nums1 []int, nums2 []int, k int) (ans int) {
	for _, i := range nums1 {
		for _, j := range nums2 {
			if i % (j * k) == 0 {
				ans++
			}
		}
	}
	return
}

File: 3162/main.rs
struct Solution {}

impl Solution {
    pub fn number_of_pairs(nums1: Vec<i32>, nums2: Vec<i32>, k: i32) -> i32 {
        let mut ans = 0;
        for i in nums1.iter() {
            for j in nums2.iter() {
                if i % (j * k) == 0 {
                    ans += 1;
                }
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::number_of_pairs(vec![1,3,4], vec![1,3,4], 1))
}